cdap-master/src/main/java/co/cask/cdap/master/startup/MasterStartupTool.java

```diff
@@ -74,6 +78,25 @@ public static void main(String[] args) {
       throw Throwables.propagate(e);
     }
 
+    // check for potential port conflicts
```

Comment:
though I guess you could still log it as a warning like now, and just not fail the check.